common/djangoapps/student/management/commands/bulk_change_enrollment.py

```diff
@@ -83,15 +83,27 @@ def handle(self, *args, **options):
         course_key_str = unicode(course_key)
 
         try:
+            course_enrollments = CourseEnrollment.objects.filter(course_id=course_key, mode=from_mode)
+            logger.info(
+                'Moving %d users from %s to %s in course %s.',
+                course_enrollments.count(), from_mode, to_mode, course_key_str
+            )
+            if not commit:
+                logger.info('Dry run, changes have not been saved. Run again with "commit" argument to save changes')
+                raise Exception('The --commit flag was not given; forcing rollback.')
             with transaction.atomic():
-                queryset = CourseEnrollment.objects.filter(course_id=course_key, mode=from_mode)
-                logger.info(
-                    'Moving %d users from %s to %s in course %s.', queryset.count(), from_mode, to_mode, course_key_str
-                )
-                queryset.update(mode=to_mode)
-
-                if not commit:
-                    raise Exception('The --commit flag was not given; forcing rollback.')
-                logger.info('Finished moving users from %s to %s in course %s.', from_mode, to_mode, course_key_str)
+                # Solution 1: With `change_enrollment`, This will also emit event.
+                for enrollment in course_enrollments:
+                    enrollment.change_mode(mode=to_mode)
```

Comment:
right — that was a concern that came up, but because this is a management command, and not run in a request process, we thought we could take a performance hit